cogs/greetings.py

```diff
@@ -28,10 +28,11 @@ async def welcome(self, ctx):
 
             message = await channel.send(embed=embed, files=[logo_file, gif_file])
             await message.add_reaction("👍")
+            self.welcome_message_id = message.id
 
     @commands.Cog.listener()
     async def on_raw_reaction_add(self, payload):
-        if payload.channel_id == constantvariables.welcome_channel_id:
+        if payload.channel_id == constantvariables.welcome_channel_id and payload.message_id == self.welcome_message_id:
```

Comment:
nit: better to put that conditional before the other one since it is much less likely that the message_id will match

However, seeing a potential failure point here. If two people join back to back without the first person reacting to the message yet, since we're doing a message per person, then if the first person reacts late, this won't run as expected right? Am I missing something here?